.github/workflows/schemas.yml

```diff
@@ -0,0 +1,73 @@
+name: 📄 Schemas
+
+on:
+  push:
+    branches: [main]
+    # branches: [main, "**"]
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      rust: ${{ steps.filter.outputs.rust }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            rust:
+              - 'homestar-invocation/src/**'
+              - 'homestar-runtime/src/**'
+              - 'homestar-schemas/src/**'
+              - 'homestar-workflow/src/**'
+
+  schemas:
+    needs: changes
+    if: ${{ needs.changes.outputs.rust == 'true' }}
+    env:
+      SCCACHE_GHA_ENABLED: "true"
+      RUSTC_WRAPPER: "sccache"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.HOMESTAR_UPDATE_TOKEN }}
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
yep, this won't be pull_request though when it goes off main. Think, it will be `push` but double check.

Reply:
It looks like we can drop the `ref` field in this case. The [`actions/checkout@v4`](https://github.com/actions/checkout?tab=readme-ov-file#usage) action says about `ref`:

> The branch, tag or SHA to checkout. When checking out the repository that
> triggered a workflow, this defaults to the reference or SHA for that event.
> Otherwise, uses the default branch.

So in our case, the SHA for the event should be the commit pushed to `main` that we are attempting to generate schemas from. Seem right?